cpp/src/arrow/ipc/reader.h

```diff
@@ -317,7 +317,12 @@ class ARROW_EXPORT Listener {
 /// \since 0.17.0
 class ARROW_EXPORT CollectListener : public Listener {
  public:
-  CollectListener() : schema_(), filtered_schema_(), record_batches_(), metadatas_() {}
+  CollectListener(bool copy_record_batch = false)
```

Comment:
I'm not sure this is a good API...
Should we create a new `CopyCollectListener` or something instead?